charts/node/templates/statefulset.yaml

```diff
@@ -17,10 +17,31 @@ spec:
   serviceName: {{ $fullname }}
   template:
     metadata:
-      {{- with .Values.podAnnotations }}
+      {{- if or .Values.podAnnotations .Values.node.vault.keys .Values.node.vault.nodeKey }}
       annotations:
+      {{- with .Values.podAnnotations }}
       {{- toYaml . | nindent 8 }}
       {{- end }}
+      {{- range $keys := .Values.node.vault.keys }}
+        vault.hashicorp.com/agent-inject-secret-{{ .name }}: {{ .vaultPath | squote }}
+        vault.hashicorp.com/agent-inject-template-{{ .name }}: |
+          {{`{{ with secret "`}}{{ .vaultPath }}{{`" }}{{- range $k, $v := .Data.data }}
+          {{ $k }}: {{ $v }}
+          {{- end }}{{ end }}`}}
```

Comment:
I am not particularly proud of this but it was the easiest way of getting values out of the secrets without having to evaluate a map inside our init container.